test/syncUserFollows.spec.js

```diff
@@ -0,0 +1,133 @@
+const env = require('./helpers/env');
+const mockAPI = env.USE_MOCK_API;
+const expectOwnProperties = require('./helpers/expectExtensions').expectOwnProperties;
+const baseUrl = config.MYFT_API_URL;
+const extraParams = `?noEvent=${config.MYFT_NO_EVENT}&waitForPurge=${config.MYFT_WAIT_FOR_PURGE_ADD}`;
+const syncUserFollowers = proxies.syncUserFollowers;
+const uuidv4 = require('uuid/v4');
+const syncUserFollowsFix = require('./mocks/fixtures/syncUserFollows');
+const syncConceptFollowsFix = require('./mocks/fixtures/syncConceptFollows');
+const edpFix = require('./mocks/fixtures/emailDigestPreference.json');
+
+const myftConst = config.myftClientConstants;
+const suppressLogs = true; //for local test if you want logs when test are run
+//TODO spec out tests for syncUserFollowers
+//TODO write stubs for:
+
+describe('syncUserFollowers', () => {
+  const fakeGroupId = '00000000-0000-0000-0000-000000000123';
+  const fakeUserId = '00000000-0000-0000-0000-000000000001';
+  const uuid = env.USER_UUID;
+  const groupId = env.LICENCE_UUID;
+
+  const followProps = {
+    byTool:"KAT",
+    byUser:"mock-admin-user"
+  };
+  let getGroupConceptStub;
+  let getGroupSyncedConceptStub;
+  let getConceptsFollowedByUserStub;
+  let addConceptsFollowedByKatUserStub;
+  let getEDPStub;
+  let noGroupConceptsToFollowStub;
+
+  let logMessageStub;
+  const logMessages = [];
+
+  before(done => {
+    if(suppressLogs) {
+      logMessageStub = sinon.stub(logger, 'log').callsFake((...params) => {
+        logMessages.push(params);
+      });
+    }
+
+    done();
+  });
+
+  beforeEach((done) => {
+    //myFT.getConceptsFollowedByGroup
+    getConceptsFollowedByUserStub = sinon.stub(myFT, 'getConceptsFollowedByUser').resolves(syncConceptFollowsFix.userConcepts);
+    addConceptsFollowedByKatUserStub = sinon.stub(myFT, 'addConceptsFollowedByKatUser').resolves(true);
+    getEDPStub = sinon.stub(myFT, 'getEmailDigestPreference').resolves(edpFix);
+    done();
+  });
+
+  afterEach(done => {
+      nock.cleanAll();
+      getConceptsFollowedByUserStub.restore();
+      addConceptsFollowedByKatUserStub.restore();
+      getEDPStub.restore();
+      done();
+  });
+
+  after(done => {
+
+    if(suppressLogs) {
+      logMessageStub.restore();
+    }
+    done();
+  });
+
+  //Happy empty path
+  it('should return status synchronisationIgnored and reason noGroupConceptsToFollow in object for no topics', (done)=> {
+    noGroupConceptsToFollowStub = sinon.stub(myFT, 'getConceptsFollowedByGroup').resolves(syncConceptFollowsFix.noGroupConcepts);
+
+    syncUserFollowers(fakeGroupId, fakeUserId).then(res => {
+      expect(res).to.be.an('object');
+      expect(res).to.have.deep.property('user.status', 'synchronisationIgnored');
+      expect(res).to.have.deep.property('user.reason', 'noGroupConceptsToFollow');
+
+    }).then(() => {
+      //clean up
+      noGroupConceptsToFollowStub.restore();
+      done();
+    }).catch(done);
```

Comment:
Also, you can get rid of all this `done` crap and having to `catch` if you just return the original promise from the test.

Reply:
Can do, but have found it useful to have the done, catch  for when a promise does fail so I get something more useful than 'timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.'  Happy to hear a better way of doing this.